src/Common/src/System/Net/Internals/IPEndPointExtensions.cs

```diff
@@ -9,28 +9,63 @@ internal static class IPEndPointExtensions
     {
         public static Internals.SocketAddress Serialize(EndPoint endpoint)
         {
-            Debug.Assert(endpoint is IPEndPoint);
```

Comment:
Based on your comment below:
> There are already existing tests for DnsEndpoint. The code checks if it's DNS end point before it calls IPEndPointExtensions.

Please update this assert instead of removing it.